src/rpcdataacq.cpp

```diff
@@ -40,104 +40,139 @@ UniValue rpc_getblockstats(const UniValue& params, bool fHelp)
         throw runtime_error(
             "getblockstats mode [startheight [endheight]]\n"
             "\n"
-            "Show stats on what wallets and cpids staked recent blocks.\n");
-    long mode= params[0].get_int();
-    (void)mode; //TODO
-    long lowheight= 0;
-    long highheight= INT_MAX;
-    long maxblocks= 14000;
-    if (mode==0)
+            "Show stats on what wallets and cpids staked recent blocks.\n"
+            "\n"
+            "Mode 0: Startheight is the starting height, endheight is the chain head if not specfied.\n"
+            "Mode 1: Startheight is actually the number of blocks back from endheight or the chain \n"
+            "        head if not specified.");
+
+    unsigned int mode = params[0].get_int();
+
+    int64_t lowheight = 0;
+
+    // Even though these are typed to int64_t for the future, the max here is set to the max for the integer type,
+    // because CBlockIndex nHeight is still type int.
+    int64_t highheight = INT_MAX;
```

Comment:
I saw that you swapped `INT_MAX` with `std::numeric_limits<int64_t>::max()` below. Since you changed these types, do you think we should do that for the rest of these?

Reply:
Yep. We should be using the built in fromt the standard rather than the macro stuff... I will do it before merging.

Reply:
For the block ones, I used <int> even though it is cast on assignment to int64_t, because of the type in CBlockIndex, and the int64_t for directly compatibility with UniValue.